src/Razor/benchmarks/Microsoft.AspNetCore.Razor.Microbenchmarks/LanguageServer/RazorDiagnosticsBenchmark.cs

```diff
@@ -44,10 +47,14 @@ public enum FileTypes
     public async Task SetupAsync()
     {
         var languageServer = RazorLanguageServer.GetInnerLanguageServerForTesting();
-
-        DocumentPullDiagnosticsEndpoint = new DocumentPullDiagnosticsEndpoint(
-            languageServerFeatureOptions: languageServer.GetRequiredService<LanguageServerFeatureOptions>(),
-            translateDiagnosticsService: languageServer.GetRequiredService<RazorTranslateDiagnosticsService>(),
+        var languageServerFeatureOptions = BuildFeatureOptions();
+        var razorDocumentMappingService = BuildRazorDocumentMappingService();
+        var loggerFactory = BuildLoggerFactory();
+        var translateDiagnosticsService = new RazorTranslateDiagnosticsService(razorDocumentMappingService, loggerFactory);
+
+        _documentPullDiagnosticsEndpoint = new DocumentPullDiagnosticsEndpoint(
+            languageServerFeatureOptions: languageServerFeatureOptions,
+            translateDiagnosticsService: translateDiagnosticsService,
             languageServer: new ClientNotifierService(BuildDiagnostics(N)), telemetryReporter: null);
```

Comment:
Consider moving `telemetryReported: null` to a separate line to stay consistent.

```suggestion
            languageServer: new ClientNotifierService(BuildDiagnostics(N)),
            telemetryReporter: null);
```